sklearn/tree/_utils.pyx

```diff
@@ -25,17 +24,19 @@ np.import_array()
 # Helper functions
 # =============================================================================
 
-cdef realloc_ptr safe_realloc(realloc_ptr* p, size_t nelems) except *:
+cdef realloc_ptr safe_realloc(realloc_ptr* p, size_t nelems) nogil except *:
```

Comment:
I'm not able to use `except NULL` because of the `realloc_ptr` return trick... But I think `except *` in this place alone should be fine?